### PR TITLE
QueryVariableEditor: Select a variable ds does not work

### LINKED ISSUE #82984: Variable using a data source from another variable saves correctly, but Variable UI doesn't display properly
### What happened?

https://github.com/grafana/grafana/assets/3722105/12a3d79f-a7f2-492f-b78e-53e619a0a4b1

### What did you expect to happen?

I would expect that the new "status" variable would show "${Prometheus}" as the data source.  This seems to work in Grafana 10.3 as expected, so this appears to be related to 10.4.

### Did this work before?

10.3 appears to work as expected

### How do we reproduce it?

See video

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: Cloud 10.4
OS: Ubuntu 22.04
Browser: Version 121.0.6167.139 (Official Build) (64-bit)

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

Prometheus

### PR DESCRIPTION
**Pull request by ivanortegaalba**
Fixes #82984

**Problem**
To use DS variables, we cannot resolve the value before passing it to the DS picker. Instead, we should move only the reference, to have the proper value selected.

This is why, in the video reproducing the bug on the issue attached to it, we see that after selecting the variable on the picker, we resolve it and pass it down to the DS picker, so the resolved DS from the DS variable selected is the one displayed.

**Solution**
Manage DS ref only. Adapt old and new variable editors to do that

### CODE REVIEW

File: public/app/features/variables/query/QueryVariableEditor.tsx [hunk 1]
```diff
@@ -130,8 +131,7 @@ export class QueryVariableEditorUnConnected extends PureComponent<Props, State>
       return null;
     }
 
-    const datasource = extended.dataSource;
-    const VariableQueryEditor = extended.VariableQueryEditor;
+    const datasource: DataSourceRef = { uid: variable.datasource?.uid, type: variable.datasource?.type };
```
Comment (torkelo) [inline public/app/features/variables/query/QueryVariableEditor.tsx:134]: is variable.datasource not already a DataSourceRef?

File: public/app/features/dashboard-scene/settings/variables/components/QueryVariableForm.tsx [hunk 2]
```diff
@@ -40,12 +41,11 @@ interface QueryVariableEditorFormProps {
 }
 
 export function QueryVariableEditorForm({
-  datasource,
+  datasource: datasourceRef,
```
Comment (torkelo) [inline public/app/features/dashboard-scene/settings/variables/components/QueryVariableForm.tsx:44]: I would rename the prop to dsRef instead to make it more clear (and not have to rename prop locally here)

**Review by torkelo — APPROVED** [review]
Looks good

**Review by oscarkilhed — APPROVED** [review]
Tested it out. Works fine. Had the same nit as Torkel about the property name for datasource.

File: public/app/features/variables/query/QueryVariableEditor.tsx (hunk 1 above)
Comment (ivanortegaalba) [reply public/app/features/variables/query/QueryVariableEditor.tsx:134]: Ups, you're right, I was getting it before from extended, which is a DataSourceAPI, and I left it like that, good catch Torkel!

File: public/app/features/dashboard-scene/settings/variables/components/QueryVariableForm.tsx (hunk 2 above)
Comment (ivanortegaalba) [reply public/app/features/dashboard-scene/settings/variables/components/QueryVariableForm.tsx:44]: We're using `datasource` for other different components(`QueryEditorRows`, `AdHocFilterValue`,`AdHocFilterKey`,`AdHocFilterEditor`), including the `QueryVariableModel`, this is why I decided to use `datasource`. We're using [the naming/renaming](https://github.com/search?q=repo%3Agrafana%2Fgrafana%20datasource%3A%20datasourceRef%2C&type=code) in different places as well.

This is why I preferred `datasource`.

**Pull request merged**